src/main/java/fi/metatavu/metamind/story/StoryController.java

```diff
@@ -340,6 +355,18 @@ public void deleteStory(Story story) {
    * @param knot knot
    */
   public void deleteKnot(Knot knot) {
+    if (knotIntentModelDAO.findByKnotAndType(knot, TrainingMaterialType.INTENTOPENNLPDOCCAT) != null) {
```

Comment:
Change to use list by knot, instead of limiting by the type

Reply:
Changed